tmt/base.py

```diff
@@ -4633,7 +4662,7 @@ def save(self) -> None:
             root=str(self.tree.root) if self.tree.root else None,
             plans=[plan.name for plan in self._plans] if self._plans is not None else None,
             steps=list(self._cli_context_object.steps),
-            environment=self.environment,
+            environment=self._environment_from_cli,
```

Comment:
If I understand this one will only save the cli sources, while previously it would have accumulated the environment from the previous runs from the `run.yaml`?